app/store.go

```diff
@@ -325,6 +314,16 @@ func (s *StoreApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBegi
 	ctx = weave.WithHeight(ctx, req.Header.GetHeight())
 	s.blockContext = ctx
 
+	if s.chainID != "" {
+		return
+	}
+
+	err := s.storeChainID(req.Header.ChainID)
```

Comment:
storeChainID updates baseContext.
The lines above create the blockContext from the baseContext.

As written the chainID will be empty on the first blokc... I would move this above top the top within a `if s.chainID == ""` guard